src/ROOTWriter.cc

```diff
@@ -175,4 +195,11 @@ ROOTWriter::checkConsistency(const std::vector<std::string>& collsToWrite, const
   return {std::vector<std::string>{}, collsToWrite};
 }
 
+void ROOTWriter::fillParams(CategoryInfo& catInfo, const GenericParameters& params) {
+  catInfo.intParams = {params.getKeys<int>(), params.getValues<int>()};
+  catInfo.floatParams = {params.getKeys<float>(), params.getValues<float>()};
+  catInfo.doubleParams = {params.getKeys<double>(), params.getValues<double>()};
+  catInfo.stringParams = {params.getKeys<std::string>(), params.getValues<std::string>()};
```

Comment:
Technically we should only take one lock here per type, rather than doing it separately for the keys and values.